lib/src/runner/reporter/xunit.dart

```diff
@@ -0,0 +1,363 @@
+    if (state.status != Status.complete) {
+      if (state.status == Status.running) {
+        if (!_testCases.containsKey(liveTest.test)) {
+          String testName = liveTest.individualName;
+          testName = testName.replaceAll('&', '&amp;');
+          testName = testName.replaceAll('<', '&lt;');
+          testName = testName.replaceAll('>', '&gt;');
+          testName = testName.replaceAll('"', '&quot;');
+          testName = testName.replaceAll("'", '&apos;');
+
+          _testCases[liveTest] = new XunitTestResult(
+              testName, liveTest.suite.path, _stopwatch.elapsedMilliseconds);
+          List listOfGroups = _orderedGroupList(liveTest.groups);
+          Map listGroup = _groupStructure;
+
+          listOfGroups.forEach((elem) {
+            if (!listGroup.containsKey(elem)) {
+              listGroup[elem] = {'testResults': new XunitTestSuite()};
+            }
+            listGroup = listGroup[elem];
+          });
+
+          listGroup['testResults'] ??= new XunitTestSuite();
+
+          listGroup['testResults'].add(_testCases[liveTest]);
+        }
+      }
+      return;
+    }
+
+    _testCases[liveTest].endTime = _stopwatch.elapsedMilliseconds;
+
+    if (liveTest.test.metadata.skip) {
+      _testCases[liveTest].skipped = true;
+    }
+
+    if (liveTest.test.metadata.skipReason != null) {
+      _testCases[liveTest].skipReason = liveTest.test.metadata.skipReason;
+    }
+  }
+
+  /// A callback called when [liveTest] throws [error].
+  void _onError(LiveTest liveTest, error, StackTrace stackTrace) {
+    if (liveTest.state.status != Status.complete) return;
+    if (_testCases[liveTest].error.isEmpty) {
+      if (error is TestFailure) {
+        _failureCount++;
+        _testCases[liveTest].error['failure'] = true;
+      } else {
+        _errorCount++;
+        _testCases[liveTest].error['failure'] = false;
+      }
+    }
+    String body = terseChain(stackTrace).toString().trim();
+    body = body.replaceAll('<', '&lt;');
+    body = body.replaceAll('>', '&gt;');
+    String name = error.toString().replaceAll('\n', '');
+    _testCases[liveTest].error['list'] ??= [];
+    _testCases[liveTest].error['list'].add(new XunitFailureResult(name, body));
+  }
+
+  /// A method used to format individual testcases
+  String _formatTestResults(List list, int counter) {
+    String testResults = '';
+    list.forEach((XunitTestResult test) {
+      String individualTest = '';
+      if (test.error.isEmpty) {
+        individualTest +=
+            '<testcase classname=\"${test.path}\" name=\"${test.name}\" time=\"${test.endTime - test.beginningTime}\"> </testcase>';
+      } else {
+        individualTest +=
+            '<testcase classname=\"${test.path}\" name=\"${test.name}\">';
+        if (test.error['failure']) {
+          test.error['list'].forEach((XunitFailureResult testFailure) {
+            individualTest += '\n' +
+                ' ' * (counter + 4) +
```

Comment:
since counter is variable, maybe you instead make constants for the various indentation strings:
```
const String indent4 = '    ';
...
... + indent4 * counter + ...
```